File: Practica_5/civitasJava/src/civitas/Sorpresa.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;
 
import java.util.ArrayList;

/**
 * @author Marina Muñoz Cano y Mario López González
 */
public abstract class Sorpresa {

    private String texto;
    private int valor;
    private Tablero tablero;

    Sorpresa(String texto, int valor, Tablero tablero) {
        this.texto = texto;
        this.valor = valor;
        this.tablero = tablero;
    }

    abstract void aplicarAJugador(int actual, ArrayList<Jugador> todos);

    /*
    Método informe()
    Parámetros:
      actual: indice del jugador actual
      todos: array que contiene los jugadores
    Efecto:
      Informa al diario que se está aplicando una sorpresa al jugador actual
     */
    protected void informe(int actual, ArrayList<Jugador> todos) {
        Diario.getInstance().ocurreEvento("Se está aplicando una sorpresa al jugador " + todos.get(actual).getNombre());
    }


    /*
    Método jugadorCorrecto()
    Parámetros:
      actual: indice del jugador actual
      todos: array que contiene los jugadores
    Return:
      Devuelve true si el jugador es correcto
     */
    public boolean jugadorCorrecto(int actual, ArrayList<Jugador> todos) {
        return (actual >= 0 && actual < todos.size());
    }

    //Public para GUI
    public String getTexto() {
        return texto;
    }

    protected int getValor(){
      return valor;
    }

    protected Tablero getTablero(){
      return tablero;
    }

}

File: EXAMEN/EXAMEN 2/civitasRuby/lib/sorpresa.rb
# encoding:utf-8
 
module Civitas
  class Sorpresa
    
    attr_reader :texto, :valor, :tablero
    
    def initialize(texto, valor=-1, tablero=nil)
      @texto = texto
      @valor = valor
      @tablero = tablero
    end
    
    def aplicar_a_jugador(iactual, todos)
    end

    # Método informe
    # Parámetros:
    #   actual: indice del jugador actual
    #   todos: array que contiene los jugadores
    # Efecto:
    #   Informa al diario que se está aplicando una sorpresa al jugador actual
    def informe(actual, todos)
      Diario.instance.ocurre_evento("Se esta aplicando una sorpresa al jugador #{todos[actual].nombre}")
    end

    # Método jugador_correcto
    # Parámetros:
    #   actual: indice del jugador actual
    #   todos: array que contiene los jugadores
    # Return:
    #   Devuelve true si el jugador es correcto
    def jugador_correcto(actual, todos)
      return (actual>=0 && actual<todos.size)
    end

    #----------------------------------------------
     protected :informe, :texto, :valor, :tablero
     private_class_method :new
    
  end
end
